### PR TITLE
feat: [PIPE-22045]: search files in repo/code explorer

### PR DESCRIPTION
**Pull request by shaurya-harness**
adds search implementation to file explorer
adds padding top to codeEditor and diffEditor

Screenshot 2024-10-30 at 4 44 17 PM

### CODE REVIEW

File: apps/gitness/src/pages/repo/repo-files.tsx [hunk 1]
```diff
@@ -21,24 +120,41 @@ export const RepoFiles: React.FC = () => {
...
   const { data: branches } = useListBranchesQuery({
     repo_ref: repoRef,
-    queryParams: { include_commit: false, sort: 'date', order: 'asc', limit: 20, page: 1, query: '' }
+    queryParams: {
+      include_commit: false,
+      sort: 'date',
+      order: 'asc',
+      limit: 20,
+      page: 1,
+      query: ''
```
Comment (cjlee01) [inline apps/gitness/src/pages/repo/repo-files.tsx:136]: when you add query to this, probably add a debounce so it doesnt rerender multiple times on input change

**Comment by cjlee01** [comment]
what do you show when there's an empty search result? @shaurya-harness

**Comment by shaurya-harness** [comment]
> what do you show when there's an empty search result? @shaurya-harness
@cjlee01 
Screenshot 2024-11-04 at 7 14 06 PM

File: apps/gitness/src/pages/repo/repo-files.tsx (hunk 1 above)
Comment (shaurya-harness) [reply apps/gitness/src/pages/repo/repo-files.tsx:136]: we are not changing this query anywhere as of now

**Comment by bradrydzewski** [comment]
Note that this should be a Textbox (with a search icon) as opposed to a Dropdown.

**Review by vardanbansal-harness — APPROVED** [review]

**Comment by shaurya-harness** [comment]
> Note that this should be a Textbox (with a search icon) as opposed to a Dropdown.

yes Brad, will be changing this in a follow up PR

**Pull request merged**